samples/cpp/facedetect.cpp

```diff
@@ -122,36 +114,27 @@ int main( int argc, const char** argv )
         if(image.empty()) cout << "Couldn't read ../data/lena.jpg" << endl;
     }
 
-    cvNamedWindow( "result", 1 );
+    namedWindow( "result", 1 );
 
-    if( capture )
+    if( capture.isOpened() )
     {
-        cout << "In capture ..." << endl;
+        cout << "Video capturing has been started ..." << endl;
+
         for(;;)
         {
-            IplImage* iplImg = cvQueryFrame( capture );
-            frame = cv::cvarrToMat(iplImg);
+            capture >> frame;
             if( frame.empty() )
                 break;
-            if( iplImg->origin == IPL_ORIGIN_TL )
-                frame.copyTo( frameCopy );
-            else
-                flip( frame, frameCopy, 0 );
 
-            detectAndDraw( frameCopy, cascade, nestedCascade, scale, tryflip );
+            detectAndDraw( frame, cascade, nestedCascade, scale, tryflip );
```

Comment:
Just to be sure you are accessing a clean copy of the data I would add a `frame.clone()` here, which is actually always a good habbit instead of just sharing pointers when grabbing from a webcam.